examples/flow-counter/interfaces/react-redux.js

```diff
@@ -0,0 +1,23 @@
+import type { Component } from 'react';
+import type { Dispatch, Store } from 'redux';
+
+
+declare module 'react-redux' {
+  declare type MapStateToProps<State, Props> = (state: State) => Props;
+  declare type MapDispatchToProps<Dispatchable, Props> = (dispatch: Dispatch<Dispatchable>) => Props;
+
+  declare function connect<State, Dispatchable, StateProps, ComponentProps, DispatchProps: $Diff<ComponentProps, StateProps>, ComponentDefaultProps, ComponentState>(
+    mapStateToProps: MapStateToProps<State, StateProps>,
+    mapDispatchToProps: MapDispatchToProps<Dispatchable, DispatchProps>
+  ) : (component: ReactClass<ComponentDefaultProps, ComponentProps, ComponentState>) => ReactClass<ComponentDefaultProps, ComponentProps, ComponentState>;
```

Comment:
Here's the fix to make `connect` type check. Using `ReactClass` instead of `Component` or `ReactComponent`. The `$Diff` internal type works, though it is still internal, so we should pay attention to the evolution of it in Flow